### DISCUSSION TITLE
Behavior of `mplhep.histplot(..., yerr=None)` different than expected

### DISCUSSION THREAD

**Discussion by matthewfeickert** [scikit-hep/mplhep#341] [Q&A]
If I use the attached `example.root` (which has to be zipped up for GitHub)

[example.zip](https://github.com/scikit-hep/mplhep/files/6933645/example.zip)

with `mplhep`, the following code

```python
import matplotlib.pyplot as plt
import uproot
import mplhep

root_file = uproot.open("example.root")
root_hists = [root_file[key].to_hist() for key in root_file.keys()]
example_hist = root_hists[-2]

fig, ax = plt.subplots()
example_hist.plot(histtype="step", yerr=None, ax=ax)
# mplhep.histplot(example_hist, histtype="step", yerr=None, ax=ax)  # equivalent
ax.set_ylim(top=30)
ax.set_title("yerr=None")
fig.savefig("plot_step_yerr_None.png", facecolor="white")

fig, ax = plt.subplots()
example_hist.plot(histtype="step", yerr=0)
# mplhep.histplot(example_hist, histtype="step", yerr=0, ax=ax)
ax.set_ylim(top=30)
ax.set_title("yerr=0")
fig.savefig("plot_step_yerr_0.png", facecolor="white")
```

produces

* `plot_step_yerr_None.png`:
![plot_step_yerr_None](https://user-images.githubusercontent.com/5142394/128236366-636c5987-3746-47fc-b666-aea006b26304.png)
* `plot_step_yerr_0.png`:
![plot_step_yerr_0](https://user-images.githubusercontent.com/5142394/128236373-041d4ab1-0342-4ac4-b413-7badce4b35d5.png)

This behavior seems a bit odd, as it seems that setting `yerr=None` should allow you to get a histogram without Poisson uncertainties drawn on it. There are times when you just want to be able to plot the histogram bin counts as the uncertanties aren't meaningful, or you're interested in the shape.

Is this behavior expected?

**Comment by andrzejnovak** [comment]
Sorry for the delay @matthewfeickert. I would say this behaviour is actually as intended as a consequence of the large set of types that can go into `histplot`. The example above will work as expected if passing explicitly `yerr=False`, `yerr=None` is taken to mean "automatic" i.e. if w2 is available yerrs will be computed and plotted from that.

The logic should be 
- array input
  - `yerr=True` - poisson errorbars
  - `yerr=False/None` - same thing, no errorbars
  - `yerr=array` - explicit errorbars
  - `w2=array` - poisson errorbars from weights
  -  `w2=array` and `yerr=array` should throw an error, hence the `None` default value
  
-  hist input
   - `w2` should always be available so should be equivalent to `w2=array`
   - can be forced off by passing `yerr=False` at the same time

I think this is reasonable, but let me know if you have any suggestions to make the API more intuitive.
